Learning Howto - Reverse Proxy + custom URL + GSSAPI in war config

AndLLA opened the pull request:
Hallo,
I created a new pull request, as the previous one was closed.
https://github.com/rundeck/docs/pull/1061

I hope I followed the suggestion in the correct way.

Thanks

AndLLA committed:
Apache 2.4 Reverse Proxy + custom URL + GSSAPI

AndLLA committed:
Apache 2.4 Reverse Proxy + custom URL + GSSAPI - menu location

fdevans commented on docs/.vuepress/sidebar-menus/learning.js:
```diff
@@ -45,7 +45,8 @@ module.exports = [{
               '/learning/howto/learn-rd-cli.md',
-              '/learning/howto/acl_basic_examples.md'
+              '/learning/howto/acl_basic_examples.md',
+              '/learning/howto/apache2-proxy-gssapi.md',              
```
remove trailing comma or the menus may not build successfully

fdevans replied:
Did you happen to test the build?

fdevans commented on docs/learning/howto/apache2-proxy-gssapi.md:
```diff
@@ -0,0 +1,104 @@
+We assume that inside the VM:
+* Rundeck has been already set up with all the necessary configuration options for DB, logging, etc.
+* apache2 is running with the necessary modules enabled (proxy, gssapi, etc)
+* a valid keytab has been generated and contains the correct principal keys (in this case HTTP/srv.example.com@EXAMPLE.COM)
```
Capital A

fdevans commented on docs/learning/howto/apache2-proxy-gssapi.md:
```diff
@@ -0,0 +1,104 @@
+is implemented by reading one file named as the user principal and located under the folder `/etc/apache2/rundeck-roles/`.
+The file contains one row with the list of roles specific to the user, for example `admin,user`.
+
+This approach is very simple and perhaps suitable only for a small user base 
```
This comment should likely go up at the top in the description of the article/solution.

fdevans reviewed (changes requested):
Had to make comments on minor changes that I'd normally handle myself, but GitHub won't let me edit your fork PR.  If you can make these small updates I'll pass it to the technical team to validate.

AndLLA committed:
fix typos

AndLLA committed:
move comment to introduction

AndLLA commented:
Hallo,
I fixed the typos and moved the comment to the introduction.
I tested the build (previously, and after the fixes) and it seems to work (note that I'm using the latest vuepress,
because some of the pinned dependencies seem to require python2, which I do not have anymore).

Thanks

![rundeck_docs_dev_run](https://user-images.githubusercontent.com/44858649/187079321-192e4ab8-6861-46f0-a0dc-24317aef1250.png)

fdevans commented:
Great, thank you.  @MegaDrive68k will help test the techincal bits and let us know his feedback soon.

MegaDrive68k reviewed (approved):
It works in my environment, all good :) Thank you @AndLLA

fdevans commented:
@AndLLA If you can resolve the last bit of conflicts I can hopefully merge before our 4.6.0 release.

AndLLA committed:
Merge branch '4.0.x' into 4.0.x-reverse-proxy-config

AndLLA commented:
Hallo,
the conflict should be fixed.

Thanks !!!
